django/middleware/locale.py

```diff
@@ -52,20 +53,20 @@ def process_response(self, request, response):
                 )
                 return self.response_redirect_class(language_url)
 
-        if not (self.is_language_prefix_patterns_used
+        if not (self.is_language_prefix_patterns_used(urlconf)
                 and language_from_path):
             patch_vary_headers(response, ('Accept-Language',))
         if 'Content-Language' not in response:
             response['Content-Language'] = language
         return response
 
-    @cached_property
-    def is_language_prefix_patterns_used(self):
+    @lru_cache.lru_cache(maxsize=None)
+    def is_language_prefix_patterns_used(self, urlconf):
```

Comment:
I could make `urlconf` default to `None` of this is deemed backward incompatible.